src/server/controllers/user.controller.ts

```diff
@@ -198,11 +202,35 @@ export const acceptTOSHandler = async ({ ctx }: { ctx: DeepNonNullable<Context>
   }
 };
 
-export const completeOnboardingHandler = async ({ ctx }: { ctx: DeepNonNullable<Context> }) => {
+export const completeOnboardingHandler = async ({
+  input,
+  ctx,
+}: {
+  input: CompleteOnboardingStepInput;
+  ctx: DeepNonNullable<Context>;
+}) => {
   try {
     const { id } = ctx.user;
-    await completeOnboarding({ id });
+    const user = await getUserById({ id, select: { onboardingSteps: true } });
+    if (!user) throw throwNotFoundError(`No user with id ${id}`);
+
+    const steps = user.onboardingSteps.filter((step) => step !== input.step);
+    const updatedUser = await updateOnboardingSteps({ id, steps });
+
+    // There are no more onboarding steps, so we can reward the user
+    if (updatedUser.onboardingSteps.length === 0) {
```

Comment:
Hm - won't evolve well for future steps, but to avoid this from being granted again, we can use the externalTransactionHack:
```js
await createBuzzTransaction({
        fromAccountId: 0,
        toAccountId: updatedUser.id,
        amount: getUserBuzzBonusAmount(ctx.user),
        description: 'Onboarding bonus',
        type: TransactionType.Reward,
        externalTransactionId: `${updatedUser.id}-onboarding-bonus`,
      });
```